db/dcrpg/upgrades.go

```diff
@@ -151,11 +157,20 @@ func updateSchemaVersion(db *sql.DB, schema uint32) error {
 	return err
 }
 
+// updater contains a number of elements necessary to perform a database
+// upgrade.
+type updater struct {
+	db      *sql.DB
+	bg      BlockGetter
+	stakeDB *stakedb.StakeDatabase
+	ctx     context.Context
+}
+
 // UpgradeDatabase attempts to upgrade the given sql.DB with help from the
 // BlockGetter. The DB version will be compared against the target version to
 // decide what upgrade type to initiate.
-func UpgradeDatabase(db *sql.DB, bg BlockGetter) (bool, error) {
-	initVer, upgradeType, err := versionCheck(db)
+func UpgradeDatabase(u *updater) (bool, error) {
```

Comment:
For this and the other functions that now take an `*updater`, I might have made the updater a receiver instead.  i.e. `func (u *updater) Upgrade() (bool, error) {`